allow most recent release with specific path

rhtenhove opened the pull request:
Wanted to install the most recent version to a specific path, but wasn't able to. Now we can!

Also fixed some edge cases:
- quotes after cp to allow paths with spaces
- allow bash elsewhere on PATH
- fixed double slash in message when installing to `/`
- don't proceed on uncaught errors, but on error do the cleanup
- Check if the given directory exists, else ask the user to create it

monopole commented:
Please sign the CL, thank.

Also,since you're working, can you improve the error on a bad argument? 
e.g.
```
# oops!
~ $ install.sh 3.1.0
curl: no URL specified!
curl: try 'curl --help' or 'curl --manual' for more information

# works!
~ $ install.sh 3.10.0
{Version:kustomize/v3.10.0 GitCommit:602ad8aa98e2e17f6c9119e027a09757e63c8bec BuildDate:2021-02-10T00:00:50Z GoOs:linux GoArch:amd64}
kustomize installed to /home/jregan/kustomize
```

monopole reviewed (commented):
Please sign the CLA per: https://github.com/kubernetes-sigs/kustomize/pull/3676#issuecomment-790742512
Thanks!

Also,since you're working on this, can you improve the error on a bad argument? 
e.g.
```
# oops!
~ $ install.sh 3.1.0
curl: no URL specified!
curl: try 'curl --help' or 'curl --manual' for more information

# works!
~ $ install.sh 3.10.0
{Version:kustomize/v3.10.0 GitCommit:3d3...bec BuildDate:2021-02-10 GoOs:linux...}
kustomize installed to /home/monopole/kustomize
```

Something that says the version doesn't exist.

In this case, it would be useful to do the curl command that reports
all the versions available for download.

rhtenhove commented:
Alright, made some changes again:
- Removed curl timeout. Seemed odd to have it wait for 10 minutes if someone forgot to plug in their ethernet cable.
- Now also checks if the target is an existing directory
- Checks if the desired version exists
- Removed `$version=""` as that doesn't do anything

rhtenhove commented:
I signed it

monopole reviewed (commented):
/lgtm

thanks!
